src/main/kotlin/de/amplimind/codingchallenge/service/UserService.kt

```diff
@@ -39,11 +39,28 @@ class UserService(
     fun fetchUserInfosForEmail(email: String): UserInfoDTO {
         return this.userRepository.findByEmail(email)?.let {
             return UserInfoDTO(
-                email = it.email,
-                isAdmin = it.role.matchesAny(UserRole.ADMIN),
-                status = extractUserStatus(it),
+                    email = it.email,
+                    isAdmin = it.role.matchesAny(UserRole.ADMIN),
+                    status = extractUserStatus(it),
+            )
+        } ?: throw ResourceNotFoundException("User with email $email was not found")
+    }
+
+    /**
+     * Deletes a user by its email
+     * @param email the email of the user to delete
+     */
+    fun deleteUserByEmail(email: String): UserInfoDTO {
+        val user = this.userRepository.findByEmail(email)
```

Comment:
schau mal wenn du das so strukturieren würdest dann sparst du dir die null checks via dem Fragezeichen

`val user = this.userRepository.findByEmail(email) ?: throw ResourceNotFoundException("User with email $email was not found")

        this.userRepository.delete(user)

        return UserInfoDTO(
                email = user.email,
                isAdmin = user.role.matchesAny(UserRole.ADMIN),
                status = extractUserStatus(user))`